framework/include/materials/MaterialBase.h

```diff
@@ -277,6 +284,14 @@ class MaterialBase : public MooseObject,
 
   virtual const QBase & qRule() const = 0;
 
+  /**
+   * Check whether a material property is active
+   */
+  bool isPropertyActive(const unsigned int prop_id) const
+  {
+    return _active_prop_ids.count(prop_id) > 0;
```

Comment:
I think this should assert that setActiveProperties has been called.
But I also dont see us introducing a new loop and forgetting to call FEProblemBase::prepareMaterial so maybe it s not useful

Reply:
I don't think it would be very useful. I think regarding properties as default inactive (or active) before a call to `setActiveProperties` is fine. Conversely, if someone calls `setActiveMaterialProperties` and then forgets to call it next time, then we would be rendering a false positive in our assertion. I just don't think we should link a check in this method to `setActiveMaterialProperties`